### PR TITLE
Improve copy-on-read sync for FastLocalCacheProvider

### PR DESCRIPTION
**Pull request by ruhan1**

### CODE REVIEW

**Comment by ruhan1** [comment]
retest this please

File: caches/infinispan/src/main/java/org/commonjava/maven/galley/cache/infinispan/FastLocalCacheProvider.java [hunk 1]
```diff
@@ -352,6 +361,10 @@ else if ( e instanceof InterruptedException )
             {
                 while ( !canStreamOpen.get() )
                 {
+                    if ( copyExceOccurs.get() )
+                    {
+                        return null;
+                    }
                     try
                     {
                         copyLock.wait();
```
Comment (jdcasey) [inline caches/infinispan/src/main/java/org/commonjava/maven/galley/cache/infinispan/FastLocalCacheProvider.java]: This wait needs to have a timeout on it so the while condition can be checked continuously. I'd set it around 100ms.

Comment (ruhan1) [reply caches/infinispan/src/main/java/org/commonjava/maven/galley/cache/infinispan/FastLocalCacheProvider.java]: I think this does not need a timeout. The main thread waits here until it is waked up by the copy thread (when either read-ready or copy-error). The while just makes sure it go to right path after it is waked up. We don't need a continuous check here.

**Comment by ruhan1** [comment]
Two tests failed. I am debugging them.

**Pull request merged**